agent/eni/iphelperwrapper/iphelper_windows_test.go

```diff
@@ -0,0 +1,275 @@
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package iphelperwrapper
+
+import (
+	"testing"
+	"time"
+	"unsafe"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	// Initial delay after which Initial notification is sent
+	InitialNotificationDelay = time.Millisecond
+	// Time delay between two successive notifications
+	TimeBetweenNotifications = 100 * time.Millisecond
+)
+
+var (
+	// Dummy function which is representative of system call (NotifyIPInterfaceChange)
+	dummyFuncNotifyIPInterfaceChange = func(...uintptr) (uintptr, uintptr, error) {
+		callback(uintptr(0), uintptr(0), uintptr(3))
+		return uintptr(0), uintptr(0), nil
+	}
+
+	// Dummy function which is representative of system call (NotifyIPInterfaceChange)
+	// Initial notification is not sent in this function
+	dummyFuncNotifyIPInterfaceChangeWithoutInitialNotification = func(...uintptr) (uintptr, uintptr, error) {
+		return uintptr(0), uintptr(0), nil
+	}
+
+	// Dummy function which is representative of system call (CancelMibChangeNotify2)
+	dummyFuncCancelMibChangeNotify2 = func(...uintptr) (uintptr, uintptr, error) {
+		return uintptr(0), uintptr(0), nil
+	}
+
+	// Dummy information for callback
+	notificationInterfaceAddition = MibIPInterfaceRow{
+		InterfaceIndex: 9,
+	}
+
+	// Dummy information for callback
+	notificationInterfaceDeletion = MibIPInterfaceRow{
+		InterfaceIndex: 9,
+	}
+)
+
+// Test for Start. In this test case, we receive an error registering the callback.
+// Therefore, our method should return with an error
+func TestStartWithErrorWhileCallingWindowsAPI(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	// Overriding the windows API call with a dummy call which is representative of system call (NotifyIPInterfaceChange)
+	// Invokation of this function returns Windows error 87 (The parameter is incorrect.)
+	funcNotifyIPInterfaceChange = func(...uintptr) (uintptr, uintptr, error) {
+		return uintptr(87), uintptr(0), nil
+	}
+	funcCancelMibChangeNotify2 = dummyFuncCancelMibChangeNotify2
+	infMonitor := NewMonitor()
+
+	err := infMonitor.Start(notificationChannel)
+	assert.Error(t, err)
+}
+
+// Test for Start. This is a success test case in which initial notification is sent immediately
+func TestStartInitialNotificationImmediately(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	// Overriding the windows API call with a dummy call
+	funcNotifyIPInterfaceChange = dummyFuncNotifyIPInterfaceChange
+	funcCancelMibChangeNotify2 = dummyFuncCancelMibChangeNotify2
+	notificationChannel := make(chan int)
+	infMonitor := NewMonitor()
+
+	err := infMonitor.Start(notificationChannel)
+
+	assert.Nil(t, err)
```

Comment:
to be consistent with assert.Error elsewhere, use assert.NoError here (and a few other places)

Reply:
I have made this change in all the files.